### PR TITLE
add gdsfactory

### PR DESCRIPTION
**Pull request by joamatab**
Checklist
- [x]  Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by joamatab** [comment]
Ready for review!

@conda-frorge/staged-recipes

**Comment by joamatab** [comment]
Ready for review 

@conda-forge/staged-recipes

**Comment by ocefpaf** [comment]
@joamatab this is not passing and it is not ready for review. It seems you need to also submit a PR to add https://pypi.org/project/lytest/, which is a dependency here.

**Comment by joamatab** [comment]
Thank you Filipe!

I updated gdsfactory where i added lytest as an optional dependency

the recipe is ready for review now

@ocefpaf 
@conda-forge/staged-recipes

**Comment by joamatab** [comment]
ready for review

**Comment by ocefpaf** [comment]
Looks like there is an entrypoint: `gf=gdsfactory.gf:gf` Maybe we can make this one noarch if we specify that in the meta.yaml.

**Comment by joamatab** [comment]
thank you Filipe! 

i added the entrypoint

@ocefpaf 
@conda-forge/staged-recipes

File: recipes/gdsfactory/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (BastianZim) [inline recipes/gdsfactory/LICENSE:1]: The license is already included in the tarball so it can be removed.

File: recipes/gdsfactory/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/gdsfactory-{{ version }}.tar.gz
+  sha256: af1bbee059a1d1cb7becbec85331a6cc899dff58d55086fa392159f93bb5b065
+
+build:
+  skip: true  # [py<37]
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - gf = gdsfactory.gf:gf
```
Comment (BastianZim) [inline recipes/gdsfactory/meta.yaml:17]: You have entry points and scripts upstream. Are both of these needed?

Ref:

Scripts: https://github.com/gdsfactory/gdsfactory/blob/5f5b045b442b372241ba2c2158e296f7f72afd11/setup.py#L37
Entry points: https://github.com/gdsfactory/gdsfactory/blob/5f5b045b442b372241ba2c2158e296f7f72afd11/setup.py#L46

Only entry points would be better as the package can then be noarch.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of small things, then this is ready to go 🚀

**Comment by joamatab** [comment]
thank you Bastian

I fixed the license and only keep the entry points

Ready for review

@BastianZim 
@conda-forge/staged-recipes

File: recipes/gdsfactory/LICENSE (hunk 1 above)
Comment (joamatab) [reply recipes/gdsfactory/LICENSE]: fixed in https://github.com/conda-forge/staged-recipes/pull/19346/commits/de9027840e81d014190a625b259452dcbf7d6167

**Comment by joamatab** [comment]
Ready for review

**Comment by joamatab** [comment]
Ready for review 

@conda-forge/staged-recipes

**Comment by joamatab** [comment]
@proppy

File: recipes/gdsfactory/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/gdsfactory/meta.yaml:17]: @joamatab did you try this suggestion? noarch packages are better b/c require less rebuilds.

Comment (joamatab) [reply recipes/gdsfactory/meta.yaml:17]: yes Filipe, i added entry points as suggested by @BastianZim 

See https://github.com/conda-forge/staged-recipes/pull/19346/files

Comment (ocefpaf) [reply recipes/gdsfactory/meta.yaml:17]: > @joamatab did you try this suggestion? noarch packages are better b/c require less rebuilds.

@joamatab we are talking about making it `noarch` now that you have the entrypoints. You will need a with `noarch: python` above the entrypoint.

Comment (joamatab) [reply recipes/gdsfactory/meta.yaml:18]: Hi Filipe,

just added `noarch: python`

https://github.com/conda-forge/staged-recipes/pull/19346/commits/cfe2b218e4086f0dcaa866d609391d6bccf06926

**Comment by BastianZim** [comment]
Please also follow the suggestion of the linter and remove the skip part.

**Comment by joamatab** [comment]
Perfect 

Thank you @BastianZim and @ocefpaf 

Ready for review

**Pull request merged**